From smurf@smurf.noris.de Mon Apr 10 20:01:56 2006
Date: Tue, 11 Apr 2006 05:01:44 +0200
To: Greg KH <gregkh@suse.de>
Cc: akpm@osdl.org, torvalds@osdl.org
Subject: USB: Overrun in option-card USB driver
Message-ID: <20060411030144.GR18658@smurf.noris.de>
Content-Disposition: inline
From: smurf@smurf.noris.de

Since the arrays are declared as in_urbs[N_IN_URB]
and out_urbs[N_OUT_URB], both for loops go one
over the end of the array. This fixes coverity id #555.

Signed-off-by: Eric Sesterhenn <snakebyte@gmx.de>
Signed-Off-By: Matthias Urlichs <smurf@smurf.noris.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -24,6 +24,7 @@
                      coding style clean-up
   2005-06-20  v0.4.1 add missing braces :-/
                      killed end-of-line whitespace
+  2006-04-10  v0.4.2 fixed two array overrun errors :-/
 
   Work sponsored by: Sigos GmbH, Germany <info@sigos.de>
 
@@ -591,14 +592,14 @@ option_setup_urbs (struct usb_serial *se
 	portdata = usb_get_serial_port_data(port);
 
 	/* Do indat endpoints first */
-	for (j = 0; j <= N_IN_URB; ++j) {
+	for (j = 0; j < N_IN_URB; ++j) {
 		portdata->in_urbs[j] = option_setup_urb (serial,
                   port->bulk_in_endpointAddress, USB_DIR_IN, port,
                   portdata->in_buffer[j], IN_BUFLEN, option_indat_callback);
 	}
 
 	/* outdat endpoints */
-	for (j = 0; j <= N_OUT_URB; ++j) {
+	for (j = 0; j < N_OUT_URB; ++j) {
 		portdata->out_urbs[j] = option_setup_urb (serial,
                   port->bulk_out_endpointAddress, USB_DIR_OUT, port,
                   portdata->out_buffer[j], OUT_BUFLEN, option_outdat_callback);
-- 
1.3-rc1.GIT

